python/ray/dag/tests/experimental/test_detect_deadlock_dag.py

```diff
@@ -0,0 +1,293 @@
+# coding: utf-8
```

Comment:
How can we make a graph with cycles with shared memory channels? For example, if I want to create a graph as follows:

```
driver -> a <-> b -> driver
```

`a` is both input and output for `b`. How can I use ADAG API to do that?

Reply:
Shared memory channels can't contribute to cycles because they are buffered.

Maybe this needs to be added to the documentation / PR description if it's not clear enough?

Reply:
@stephanie-wang, with the current ADAG API, is it possible for users to define a graph that has cycles using shared memory channels?

For example, can a data flow graph have the following edges, all of which are shared memory channels?
* driver --> a
* a --> b
* b --> a
* b --> driver

Reply:
You can create a cycle between processes (driver, actor A, actor B) but you cannot create a cycle between the tasks, i.e. a -> b -> a is only possible if the "a" nodes are different tasks on actor A.

Reply:
> Maybe this needs to be added to the documentation / PR description if it's not clear enough?

Update